custom/icds/location_reassignment/parser.py

```diff
@@ -123,3 +130,27 @@ def _validate_descendants_archived(self):
                 self.errors.append("Location %s is getting archived but the following descendants are not %s" % (
                     location.location_id, ",".join(missing_site_codes)
                 ))
+
+    def process(self):
+        # process each sheet, in reverse order of hierarchy
+        for location_type_code in list(reversed(self.location_type_codes)):
+            for operation, transitions in self.valid_transitions[location_type_code].items():
+                self._perform_transitions(operation, transitions)
+
+    def _perform_transitions(self, operation, transitions):
+        for old_site_codes, new_site_codes in transitions.items():
+            deprecate_locations(operation, self._get_locations(old_site_codes),
+                                self._get_locations(new_site_codes))
+
+    def _get_locations(self, site_codes):
+        site_codes = site_codes if isinstance(site_codes, list) else [site_codes]
+        locations = [self.locations_by_site_code[site_code] for site_code in site_codes]
+        assert len(locations), len(site_codes)
```

Comment:
Do you mean for `len(site_codes)` to be the assertion failure message? Seems more useful to use something like "Expected locations but found none."

Reply:
valid point, updated to raise exception
https://github.com/dimagi/commcare-hq/pull/26983/commits/8d11300ef844b44e80571e4b0bd6d14da71ac849